indico/modules/events/registration/controllers/management/reglists.py

```diff
@@ -602,6 +603,8 @@ class RHRegistrationCheckIn(RHManageRegistrationBase):
     """Set checked in state of a registration"""
 
     def _process_PUT(self):
+        if self.registration.state not in (RegistrationState.complete, RegistrationState.unpaid):
```

Comment:
Is this the same logic we use to hide the UI for it?

Reply:
[Yep].(https://github.com/indico/indico/blob/master/indico/modules/events/registration/templates/management/_registration_details.html#L20)